### PR TITLE
Feature/ui key binding keyvalpair

### PR DESCRIPTION
**Pull request by PGuimarais**
- Adds new row when `enter` is pressed
- Autofocuses on newly added row
- Newly added row gets placed after current row
- Add / Remove Row buttons are now selectable by tab

Bamboo Build:
http://builds.cask.co/browse/CDAP-DRC4729

### CODE REVIEW

**Comment by ajainarayanan** [comment]
🎱 🍻 👍 Awesome! LGTM

**Comment by PGuimarais** [comment]
After discussion, decided to move keypress handler from parent component to child

File: cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js [hunk 1]
```diff
@@ -13,30 +13,40 @@
...
-      <button type="submit" className="fa fa-trash remove-row-btn" onClick={removeRow} />
-    </div>
-  );
-};
+  constructor(props){
+    super(props);
+    this.keyPress = this.keyPress.bind(this);
+  }
+  keyPress(e) {
+    if(e.which === 13){
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js]: which is a deprecated API. You should use keyCode

File: cdap-ui/app/cdap/components/KeyValuePairs/index.js [hunk 2]
```diff
@@ -72,10 +73,11 @@ export default class KeyValuePairs extends Component {
     };
     this.keyValueStore = createKeyValueStore({keyValues});
     this.subscription = this.keyValueStore.subscribe(() => {
-      OnKeyValueChange(this.keyValueStore.getState().keyValues);
       this.setState(this.keyValueStore.getState().keyValues);
+      OnKeyValueChange(this.keyValueStore.getState().keyValues);
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/KeyValuePairs/index.js]: Can you change this props name to start with lower case?

File: cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js (hunk 1 above)
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js]: keyCode is actually deprecated too. Updated to use key

File: cdap-ui/app/cdap/components/KeyValuePairs/index.js (hunk 2 above)
Comment (PGuimarais) [reply cdap-ui/app/cdap/components/KeyValuePairs/index.js]: done

File: cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js (hunk 1 above)
Comment (PGuimarais) [reply cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePair.js]: Done

**Comment by elfenheart** [comment]
👍 LGTM

**Pull request merged**